service/statistics/AnalyticsEvents.js

```diff
@@ -1,447 +1,426 @@
- * getUserMedia.success.audio.video.300
- * getUserMedia.success.audio.video.360
- * getUserMedia.success.audio.video.720
- * getUserMedia.success.audio.video.960
- * getUserMedia.success.audio.video.undefined
- * getUserMedia.success.desktop
- * getUserMedia.success.video.180
- * getUserMedia.success.video.360
- * getUserMedia.success.video.720
- * getUserMedia.success.video.960
- * getUserMedia.success.video.undefined
+ * Indicates that the ICE state has changed.
  *
- * TODO: document, reformat
+ * Properties:
+ *      state: the ICE state which was entered (e.g. 'checking', 'connected',
+ *          'completed', etc).
+ *      value: the time in milliseconds (as reported by
+ *          window.performance.now()) that the state change occurred.
+ *      p2p: whether the associated ICE connection is p2p or towards a
+ *          jitsi-videobridge
+ *      signalingState: The signaling state of the associated PeerConnection
+ *      reconnect: whether the associated Jingle session is in the process of
+ *          reconnecting (or is it ICE???).
  */
-export const GET_USER_MEDIA_SUCCESS_ = 'getUserMedia.success';
+export const ICE_STATE_CHANGED = 'ice.state.changed';
 
 /**
- * Properties: none
+ * Indicates that a track was unmuted (?).
  *
- * Known full event names:
- * getUserMedia.userCancel.extensionInstall
- *
- * TODO: document, reformat
+ * Properties:
+ *      mediaType: the media type of the local track ('audio' or 'video').
+ *      type: the type of the track ('local' or 'remote').
+ *      value: ???
  */
-export const GET_USER_MEDIA_USER_CANCEL_ = 'getUserMedia.userCancel';
+export const TRACK_UNMUTED = 'track.unmuted';
 
 /**
- * Properties: value
- *
- * The "value" property contains the difference in milliseconds between
- * the ICE establishment time for the P2P and JVB connections (e.g. a value
- * of 10 would indicate that the P2P was 10ms slower than JVB).
+ * Creates an operational event which indicates that we have received a
+ * "bridge down" event from jicofo.
  */
-export const ICE_ESTABLISHMENT_DURATION_DIFF
-    = 'ice.establishmentDurationDiff';
+export const createBridgeDownEvent = function() {
+    const bridgeDown = 'bridge.down';
 
-/**
- * Properties: none
- *
- * TODO: document
- * TODO: do we need this in addition to _ICE_CONNECTION_STATE?
- */
-export const ICE_FAILED = 'connection.ice_failed';
+    return {
+        action: bridgeDown,
+        actionSubject: bridgeDown,
+        source: SOURCE_SPORADIC,
+        type: TYPE_OPERATIONAL
+    };
+};
 
 /**
- * Properties: none
- *
- * TODO: document
+ * Creates an event which indicates that the XMPP connection failed
+ * @param errorType ???
+ * @param errorMessage ???
  */
-export const P2P_ESTABLISHED = 'p2p.established';
+export const createConnectionFailedEvent = function(errorType, errorMessage) {
+    return {
+        type: TYPE_OPERATIONAL,
+        name: 'connection.failed',
+        errorType,
+        errorMessage
```

Comment:
Should errorType and errorMessage be attributes? Other actions seem to follow a strict set of keys and errorType and errorMessage aren't among them.

Reply:
Yes, nice catch!